### PR TITLE
Disambiguate condvar API contract.

### PR DESCRIPTION
**Pull request by mattmacy**
On Illumos callers of cv_timedwait and cv_timedwait_hires can't distinguish
between whether or not the cv was signalled or the call timed out. Illumos
handles this (for some definition of handles) by calling cv_signal in the return
path if we were signalled but the return value indicates instead that we timed
out. This would make sense if it were possible to query the the cv for its net
signal disposition. However, this isn't possible and, in spite of the fact that
there are places in the code that clearly take a different and incompatible
path if a timeout value is indicated, this distinction appears to be rather
subtle to most developers. This problem is further compounded by the fact
that on Linux, calling cv_signal in the return path wouldn't even do the
right thing unless there are other waiters.

Since it is possible for the caller to independently determine how much
time is remaining but it is not possible to query if the cv was in fact
signalled, prioritizing signalling over timeout seems like a cleaner
solution. In addition, judging from usage patterns within the code itself,
it is also less error prone.

Signed-off-by: Matt Macy <mmacy@FreeBSD.org>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by lundman** [comment]
So this is an API change flag day of sorts.

**Comment by mattmacy** [comment]
> So this is an API change flag day of sorts.

Correct.

**Comment by mattmacy** [comment]
> I like removal of one gethrtime() call. Thinking whether it is possible to remove another one also.
> 
> What's about KPI change, it should better be at least documented, since having the same function working differently is confusing.

My solution was to add `_spl` and return an enum. Brian prefers the smaller source compatible change.

**Comment by lundman** [comment]
Amusingly, found https://github.com/openzfs/zfs/pull/10468#discussion_r441215264
to show that macOS `spl_wait_sig()` as been returning opposite values since dawn of time, I guess nothing relied on it before.

File: module/zfs/spa.c [hunk 1]
```diff
@@ -3144,7 +3144,7 @@ spa_activity_check(spa_t *spa, uberblock_t *ub, nvlist_t *config)
 		}
 
 		error = cv_timedwait_sig(&cv, &mtx, ddi_get_lbolt() + hz);
-		if (error != -1) {
+		if (error <= 0) {
```
Comment (freqlabs) [inline module/zfs/spa.c:3147]: Why this change?
